File: src/main/java/de/danielmaile/lama/license/LicenseManager.java
package de.danielmaile.lama.license;

import org.apache.http.client.config.RequestConfig;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.client.utils.URIBuilder;
import org.apache.http.impl.client.BasicResponseHandler;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClientBuilder;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

import java.net.URI;

public record LicenseManager(JavaPlugin instance, String pluginName) {

    public LicenseManager(JavaPlugin instance, String pluginName) {
        this.instance = instance;
        this.pluginName = pluginName;

        Bukkit.getScheduler().runTaskTimerAsynchronously(instance, () -> {
            if (!validate()) {
                instance.getLogger().severe("License key validation failed! Please check your key in the config and restart the server to try again. Plugin gets disabled...");
                Bukkit.getScheduler().runTask(instance, () -> instance.getServer().getPluginManager().disablePlugin(instance));
            }
        }, 0L, 18000L); //run every 15min
    }

    private boolean validate() {
        try {
            String license = instance.getConfig().getString("license_key");
            RequestConfig config = RequestConfig.custom()
                    .setConnectTimeout(30000)
                    .setConnectionRequestTimeout(30000)
                    .setSocketTimeout(30000).build();
            CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
            HttpGet httpGet = new HttpGet("http://lamahost.de:8000/validation");
            URI uri = new URIBuilder(httpGet.getURI())
                    .addParameter("product", pluginName)
                    .addParameter("license", license)
                    .build();
            httpGet.setURI(uri);

            CloseableHttpResponse response = client.execute(httpGet);
            String responseString = new BasicResponseHandler().handleResponse(response);
            client.close();

            return Boolean.parseBoolean(responseString);
        } catch (Exception e) {
            return false;
        }
    }
}

File: src/main/java/de/danielmaile/mpp/data/LicenseManager.kt
package de.danielmaile.mpp.data

import de.danielmaile.mpp.inst
import org.apache.http.NameValuePair
import org.apache.http.client.config.RequestConfig
import org.apache.http.client.entity.UrlEncodedFormEntity
import org.apache.http.client.methods.HttpPost
import org.apache.http.impl.client.BasicResponseHandler
import org.apache.http.impl.client.HttpClientBuilder
import org.apache.http.message.BasicNameValuePair
import org.bukkit.Bukkit
import java.nio.charset.StandardCharsets
import java.util.ArrayList

object LicenseManager {

    fun validateLicense(): Boolean {
        return try {
            val config = RequestConfig.custom()
                .setConnectTimeout(10000)
                .setConnectionRequestTimeout(10000)
                .setSocketTimeout(10000).build()
            val client = HttpClientBuilder.create().setDefaultRequestConfig(config).build()
            val httpPost = HttpPost("http://sugmadezz.online:8000/validation")
            val urlParameters = getRequestInformation()
            httpPost.entity = UrlEncodedFormEntity(urlParameters, StandardCharsets.UTF_8)
            val response = client.execute(httpPost)
            val responseString = BasicResponseHandler().handleResponse(response)
            client.close()
            responseString.toBoolean()
        } catch (e: Exception) {
            false
        }
    }

    private fun getRequestInformation(): List<NameValuePair> {
        val urlParameters: MutableList<NameValuePair> = ArrayList()
        urlParameters.add(BasicNameValuePair("productName", "MPP"))
        urlParameters.add(BasicNameValuePair("licenseKey", inst().config.getString("license_key") ?: ""))
        urlParameters.add(BasicNameValuePair("serverVersion", Bukkit.getVersion()))
        urlParameters.add(BasicNameValuePair("onlinePlayers", Bukkit.getOnlinePlayers().size.toString()))
        return urlParameters
    }
}